pub/router.go

```diff
@@ -14,7 +14,8 @@ const baseIngressPath = "/pub"
 
 func ListenAndServe(cfg *Config) error {
 	gin.SetMode(gin.ReleaseMode) // Hide Debug Routings
-	router := gin.Default()
+	router := gin.New()
+	router.Use(gin.Recovery())
```

Comment:
What logs are we keeping with this change, and what logs are we throwing away?

Reply:
Take a look at the mess we got right now:
```
[GIN] 2023/03/15 - 13:32:59 | 200 |    1.351117ms |  10.233.123.253 | GET      "/metrics"
[GIN] 2023/03/15 - 13:33:08 | 200 |      55.401µs |     172.18.54.6 | GET      "/live"
[GIN] 2023/03/15 - 13:33:08 | 200 |        47.8µs |     172.18.54.6 | GET      "/ready"
[GIN] 2023/03/15 - 13:33:09 | 200 |     2.34813ms |  10.233.120.207 | GET      "/metrics"
[2023-03-15T13:33:09+0000] INFO Incoming Proxy request                   method=GET path=/pub/job/adder/0.28.2 requestId=b4ae3e2c-3624-48ca-b316-641e2ef0a675
[2023-03-15T13:33:10+0000] INFO Proxy request done                       jobName=adder jobPath=/pub/job/adder/0.28.2 jobVersion=0.28.2 requestId=b4ae3e2c-3624-48ca-b316-641e2ef0a675 status=307
[GIN] 2023/03/15 - 13:33:10 | 307 |  195.433717ms | 185.153.154.186 | GET      "/pub/job/adder/0.28.2"
[2023-03-15T13:33:10+0000] INFO Incoming Proxy request                   method=GET path=/pub/job/adder/0.28.2/ requestId=7b0b9e3f-7052-4ba0-bb8b-faac13b72fb1
[2023-03-15T13:33:10+0000] INFO Proxy request done                       jobName=adder jobPath=/pub/job/adder/0.28.2/ jobVersion=0.28.2 requestId=7b0b9e3f-7052-4ba0-bb8b-faac13b72fb1 status=200
[GIN] 2023/03/15 - 13:33:10 | 200 |  174.305645ms | 185.153.154.186 | GET      "/pub/job/adder/0.28.2/"
[2023-03-15T13:33:12+0000] INFO Incoming Proxy request                   method=GET path=/pub/job/adder/0.28.2/openapi.json requestId=67eda4b5-69b6-4bfc-972b-b3e496eb4d23
[2023-03-15T13:33:12+0000] INFO Proxy request done                       jobName=adder jobPath=/pub/job/adder/0.28.2/openapi.json jobVersion=0.28.2 requestId=67eda4b5-69b6-4bfc-972b-b3e496eb4d23 status=200
[GIN] 2023/03/15 - 13:33:12 | 200 |  197.710346ms | 185.153.154.186 | GET      "/pub/job/adder/0.28.2/openapi.json"
[GIN] 2023/03/15 - 13:33:18 | 200 |     114.701µs |     172.18.54.6 | GET      "/ready"
[GIN] 2023/03/15 - 13:33:18 | 200 |     167.802µs |     172.18.54.6 | GET      "/live"
```

We do have our custom request logs:
- `[2023-03-15T13:33:12+0000] INFO Incoming Proxy request` - when a request starts. We keep that.
- `[2023-03-15T13:33:12+0000] INFO Proxy request done` - when a request ends. We keep that.

Besides, there are duplicated request logs coming from GIN:
- `[GIN] 2023/03/15 - 13:33:12 | 200 |  197.710346ms | 185.153.154.186 | GET      "/pub/job/adder/0.28.2/openapi.json"` - that we're gonna remove

Also, removing liveness & readiness probes which produces a ton of logs:
- `[GIN] 2023/03/15 - 13:33:18 | 200 |     114.701µs |     172.18.54.6 | GET      "/ready"`

Reply:
Right, it just isn't obvious to me _what_ exactly we're throwing away and keeping. I agree with the change as per the original issue, but are we throwing away actual error messages as well, or just a bunch of debug info?

Reply:
Just so you know, `gin.Default()` was doing this under the hood:
```go
        engine := New()
	engine.Use(Logger(), Recovery())
```
namely, using 2 middlewares: Logger and Recovery.
With my change I threw away Logger middleware, but Recovery stays. I just checked that if you raise a panic, the error is still being printed out. So we're throwing away only debug info that doesn't add anything to what we already have.